pkg/verification/verification_service.go

```diff
@@ -63,13 +64,43 @@ func NewVerificationService(cfg ServiceConfiguration, opts ...VerificationServic
 
 // InitVerification sends a verification message to the specified user.  If successful,
 // the user will receive a verification SMS.
-func (s *ServiceImpl) InitVerification(ctx *gin.Context, signup *v1alpha1.UserSignup, countryCode, phoneNumber string) (*v1alpha1.UserSignup, error) {
-	// get phone number hash
+func (s *ServiceImpl) InitVerification(ctx *gin.Context, userID string, e164PhoneNumber string) error {
+	signup, err := s.Services().SignupService().GetUserSignup(userID)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Error(ctx, err, "usersignup not found")
+			errors.AbortWithError(ctx, http.StatusNotFound, err, "usersignup not found")
+			return
+		}
+		log.Error(ctx, err, "error retrieving usersignup")
+		errors.AbortWithError(ctx, http.StatusInternalServerError, err, fmt.Sprintf("error retrieving usersignup: %s", userID))
```

Comment:
same here, we could use `errors.NewInternalError(...)` but here we use `errors.AbortWithError(...)` is there a reason? or could we stick to a more unified approach?